microprofile/testing/junit5/pom.xml

```diff
@@ -54,6 +54,11 @@
             <artifactId>junit-jupiter-api</artifactId>
             <scope>provided</scope>
         </dependency>
+        <dependency>
```

Comment:
Can you consult @tomas-langer about adding this dependency.

Reply:
It is necessary to compile `helidon-microprofile-testing-junit5`. Users don't need to include mockito dependencies as long as they don't add `@MockBean`. In case they add `@MockBean` without having mockito dependency they will have a runtime error ClassNotFoundException.

I would personally remove every provided scope in the pom.

Reply:
`HelidonJunitExtension` imports mockito so it's effectively required to have it on the test class-path.
I'm basically asking if we should modularize this. 

@tomas-langer please chime in.

Reply:
Import does not matter, code path that uses the imported type matters. I will check the code.

Reply:
If the dependency needs to be added by the user (i.e. we expect the user already has it on their classpath), it should be `provided`. If it is not required to run this code, it should be `<optional>true</optional>` instead.

Reply:
I.e. we don't have to modularize, we only need to ensure that the normal code path doesn't fail because of the absence of Mockito.